supervisor/src/bin/supervisor-encore.rs

```diff
@@ -58,16 +59,27 @@ pub async fn main() {
             )
             .expect("Failed to create process for gateways"),
         );
+        port += 1;
     }
 
-    let proxy = proxy::GatewayProxy::new(
-        reqwest::Client::new(),
-        std::net::SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), port),
-        service_ports.clone(),
-    );
-    let sv = Supervisor::new(procs);
+    let mut handles = Vec::with_capacity(2);
+    let mut results = Vec::with_capacity(handles.len());
     let root_token = CancellationToken::new();
+
+    let sv = Supervisor::new(procs);
     let supervisor_token = root_token.child_token();
+    handles.push(tokio::spawn(sv.supervise(supervisor_token)));
+
+    if use_proxy {
+        let proxy = proxy::GatewayProxy::new(
+            reqwest::Client::new(),
+            std::net::SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), port - 1),
```

Comment:
Why `port-1`?

Reply:
The port is an incremental value increased after each assignment.

Reply:
But isn't that port assigned to some other proc?

Reply:
Exactly. It's the proxy target. I.e. the port the proxy will forward all requests to. The serving port is assigned in the serve method